### PR TITLE
fix(expr): align `position` syntax and `strpos` semantic with SQL standard

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The SQL standard (and PostgreSQL) defines it as `POSITION(substring IN string)` but our previous implementation was `POSITION(string, substring)`. And `PostgreSQL` deliberately does not allow the comma syntax as extension to avoid confusion of parameter order. **This is technically a breaking change to the parser. Existing materialized views can run without any changes, but queries stored in an external script or non-materialized views may fail to parse.**

PostgreSQL also has `strpos(string, substring)` sharing the same implementation. We accidentally implemented them separately, one with a wrong semantic generated by chatgpt, that returns bytes position rather than character position. This PR merges them to the same correct implementation. Although the old proto enum value is retained, this could still be considered as **breaking change because the semantic (and return value) is changing**.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

- SQL commands, functions, and operators

### Release note

Support `position` with standard SQL syntax, and fix `strpos` to return character position rather than byte position.

### CODE REVIEW

File: src/expr/src/vector_op/position.rs [hunk 1]
```diff
@@ -14,16 +14,43 @@
...
+/// 2
+///
+/// query I
+/// select strpos('abc', 'def');
+/// ----
+/// 0
 ///
-/// Note: According to pgsql, position will return 0 rather -1 when substr is not in the target str
+/// query I
+/// select strpos('床前明月光', '月光');
```
Comment (kwannoel) [inline src/expr/src/vector_op/position.rs:44]: Does the  \`\`\`slt\`\`\` block mean that all these slt cases will be tested in e2e test?

File: proto/expr.proto [hunk 2]
```diff
@@ -102,7 +102,7 @@ message ExprNode {
     LPAD = 238;
     RPAD = 239;
     REVERSE = 240;
-    STRPOS = 241;
+    STRPOS = 241 [deprecated = true]; // duplicated with POSITION
```
Comment (kwannoel) [inline proto/expr.proto:105]: Why `deprecated`, even if duplicated I thought it is permissible, just as an alias?

Comment (kwannoel) [reply proto/expr.proto:105]: Oh I see, we will just rewrite into `POSITION` in frontend, so we won't use `STRPOS` in protobuf.

File: src/expr/src/vector_op/position.rs (hunk 1 above)
Comment (xiangjinwu) [reply src/expr/src/vector_op/position.rs:44]: Yes. #5117

**Review by kwannoel — APPROVED** [review]
LGTM

File: proto/expr.proto (hunk 2 above)
Comment (xiangjinwu) [reply proto/expr.proto:105]: Consider this as dead / duplicate code cleaning. User facing alias is handled by sqlparser and binder.

**Pull request merged**